lightbug_http/sys/net.mojo

```diff
@@ -216,14 +216,14 @@ struct SysConnection(Connection):
         self.fd = fd
 
     fn read(self, inout buf: Bytes) raises -> Int:
-        var new_buf = Pointer[UInt8]().alloc(default_buffer_size)
+        var new_buf = UnsafePointer[UInt8]().alloc(default_buffer_size)
         var bytes_recv = recv(self.fd, new_buf, default_buffer_size, 0)
         if bytes_recv == -1:
             return 0
         if bytes_recv == 0:
             return 0
-        var bytes_str = String(new_buf.bitcast[Int8](), bytes_recv)
-        buf = bytes_str._buffer
+        var bytes_str = String(new_buf.bitcast[UInt8](), bytes_recv + 1)
+        buf = bytes(bytes_str, pop=False)
         return bytes_recv
 
```

Comment:
This does not handle large payloads very well in combination with the request parsing logic in the sys/server. Needs to be refactored

Reply:
You could try reading directly into the ptr owned by buf if you want to try avoiding the additional string allocation. I'm not sure how much time it would save, but that's how I've been doing it for my net file descriptor implementation. Hasn't blown up *yet*, might be worth a try: https://github.com/thatstoasty/gojo/blob/main/gojo/net/fd.mojo#L47

```mojo
fn read(inout self, inout dest: List[Byte]) -> (Int, Error):
        """Receive data from the file descriptor and write it to the buffer provided."""
        var bytes_received = recv(
            self.fd,
            DTypePointer[DType.uint8](dest.unsafe_ptr()).offset(dest.size),
            dest.capacity,
            0,
        )
        if bytes_received == -1:
            return 0, Error("Failed to receive message from socket.")
        dest.size += bytes_received

        if bytes_received < dest.capacity:
            return bytes_received, Error(io.EOF)

        return bytes_received, Error()
```

Reply:
Ah, cool! Thanks! @alainrollejr going to try this out today

Reply:
Done, seems to work nicely